src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -3,19 +3,182 @@
+        productDao.save(new Product("sec901", "Sony Ericsson C901", new BigDecimal(420), usd, 30, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Sony/Sony%20Ericsson%20C901.jpg"));
+        productDao.save(new Product(CODE_XPERIAXZ, "Sony Xperia XZ", new BigDecimal(120), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Sony/Sony%20Xperia%20XZ.jpg"));
+        productDao.save(new Product("nokia3310", "Nokia 3310", new BigDecimal(70), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Nokia/Nokia%203310.jpg"));
+        productDao.save(new Product("palmp", "Palm Pixi", new BigDecimal(170), usd, 30, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Palm/Palm%20Pixi.jpg"));
+        productDao.save(new Product(CODE_SIMC_56, "Siemens C56", new BigDecimal(70), usd, 20, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20C56.jpg"));
+        productDao.save(new Product("simc61", "Siemens C61", new BigDecimal(80), usd, 30, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20C61.jpg"));
+        productDao.save(new Product("simsxg75", "Siemens SXG75", new BigDecimal(150), usd, 40, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20SXG75.jpg"));
+        notDefaultProduct = new Product(CODE, DESCRIPTION, PRICE, usd, STOCK, IMAGE_URL);
+    }
+
+
+    @Test
+    public void shouldFindProductById() {
+        Product result = productDao.getProduct(EXIST_ID);
+
+        assertEquals((Long) EXIST_ID, result.getId());
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void shouldFindNonExistProduct() {
+        productDao.getProduct(NON_EXIST_ID);
     }
 
+
     @Test
-    public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+    public void shouldDeleteProduct() {
+        Long id = EXIST_ID;
+
+        productDao.delete(id);
+
+        long count = productDao
+                .getProducts()
+                .stream()
+                .filter(p -> id.equals(p.getId()))
+                .count();
+        assertEquals(ZERO_COUNT, count);
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void shouldSaveNull() {
+        productDao.save(null);
+    }
+
+    @Test
+    public void shouldProductWithNullId() {
+        notDefaultProduct.setId(null);
+
+        productDao.save(notDefaultProduct);
+
+        long count = productDao
+                .getProducts()
+                .stream()
+                .filter(p -> notDefaultProduct.getDescription().equals(p.getDescription()))
+                .filter(p -> p.getId() != null)
+                .count();
+        assertEquals(ONE_COUNT, count);
```

Comment:
Please separate method's invocation section and assertion section with blank line. Do it for the entire project.